tests/admin/apple-actions/index/test-class-push.php

```diff
@@ -258,22 +255,103 @@ public function testComponentErrorsFail() {
 
 		$action = new Push( $this->settings, $post_id );
 		$action->set_api( $api->reveal() );
+
+		try {
+			$action->perform();
+		} catch ( Action_Exception $e ) {
+
+			// An admin error notice was created
+			$notices = get_user_meta( $user_id, 'apple_news_notice', true );
+			$this->assertNotEmpty( $notices );
+
+			$component_notice = end( $notices );
+			$this->assertEquals( 'The following components are unsupported by Apple News and prevented publishing: iframe', $e->getMessage() );
+
+			// The post was not sent to Apple News
+			$this->assertEquals( null, get_post_meta( $post_id, 'apple_news_api_id', true ) );
+			$this->assertEquals( null, get_post_meta( $post_id, 'apple_news_api_created_at', true ) );
+			$this->assertEquals( null, get_post_meta( $post_id, 'apple_news_api_modified_at', true ) );
+			$this->assertEquals( null, get_post_meta( $post_id, 'apple_news_api_share_url', true ) );
+			$this->assertEquals( null, get_post_meta( $post_id, 'apple_news_api_deleted', true ) );
+		}
+	}
+
+	public function testJSONErrorsWarn() {
+		$this->settings->set( 'component_alerts', 'none' );
+		$this->settings->set( 'json_alerts', 'warn' );
+
+		$response = $this->dummy_response();
+		$api = $this->prophet->prophesize( '\Apple_Push_API\API' );
```

Comment:
It's part of a unit test library called Prophecy. It allows you to instantiate dummy versions of complex classes, initialize with sample data when exact values aren't relevant, check simply whether a function executes but not necessarily execute with any specific value. It's very useful.